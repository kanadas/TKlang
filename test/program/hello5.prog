
head :: [char] -> char + unit;
head l = match l with {
    h:t -> 1@h;
    [] -> 2@()
    };

stream buffer
input Input
state
    tail = tail;
    buf :: [char];
    buf = tail buf
output out = head buf
initial 
    buf = "Hello world!", 
    out = 2 @ (),
    tail l = match l with {
        h:t -> t;
        [] -> []
    };

stream main
input buffer
state
output print = match buffer.out with {
    1@c -> c;
    _ -> '0'
    }
initial print = '0'
