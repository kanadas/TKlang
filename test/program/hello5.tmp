program/hello5.prog
