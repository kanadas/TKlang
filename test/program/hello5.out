program/hello5.prog
Hello 